### PR TITLE
[Feature] init affinity matcher with array

### PR DESCRIPTION
**Pull request by huguesva**

### CODE REVIEW

**Comment by huguesva** [comment]
The function `test_array_init` in `test_neighbor_embedding.py` works when using SGD but not ADAM.
Any thoughts on this ?

File: torchdr/tests/test_neighbor_embedding.py [hunk 1]
```diff
@@ -38,6 +40,39 @@ def toy_dataset(n=300, dtype="float32"):
     return X.astype(dtype), y
 
 
+@pytest.mark.parametrize("dtype", lst_types)
+@pytest.mark.parametrize("keops", lst_keops)
+def test_array_init(dtype, keops):
+    n = 300
```
Comment (rflamary) [inline torchdr/tests/test_neighbor_embedding.py]: this is a bit large maybe decrease to 59

File: torchdr/tests/test_neighbor_embedding.py [hunk 2]
```diff
@@ -38,6 +40,39 @@ def toy_dataset(n=300, dtype="float32"):
...
+    X, y = toy_dataset(n, dtype)
+
+    Z_init_np = np.random.randn(n, 2).astype(dtype)
+    Z_init_torch = torch.from_numpy(Z_init_np)
+
+    lst_Z = []
+    for Z_init in [Z_init_np, Z_init_torch]:
+        model = TSNE(
```
Comment (rflamary) [inline torchdr/tests/test_neighbor_embedding.py]: the init should work on all estimator no? maybe do  a loop? you can decrease the number of iteration ti keep tests fast.

File: torchdr/tests/test_neighbor_embedding.py [hunk 3]
```diff
@@ -53,21 +55,55 @@ def toy_dataset(n=300, dtype="float32"):
...
+    Z = model.fit_transform(X)
+
+    check_shape(Z, (n, 2))
+    assert silhouette_score(Z, y) > 0.2, "Silhouette score should not be too low."
+
+
+@pytest.mark.parametrize("dtype", lst_types)
+@pytest.mark.parametrize("keops", lst_keops)
+def test_array_init(dtype, keops):
+    n = 300
```
Comment (rflamary) [inline torchdr/tests/test_neighbor_embedding.py:81]: zlso large no need for large data in tetss

**Pull request merged**